pkg/cmd/roachtest/testselector/selector.go

```diff
@@ -0,0 +1,205 @@
+	sfPasswordEnv = "SFPASSWORD"
+)
+
+//go:embed snowflake_query.sql
+var preparedQuery string
+
+// supported suites
+var suites = map[string]string{
+	"nightly": "roachtest nightly",
+}
+
+// TestDetails has the details of the test as fetched from snowflake
+type TestDetails struct {
+	Name                string // test name
+	Selected            bool   // whether a test is Selected or not
+	AvgDurationInMillis int64  // average duration of the test
+	TotalRuns           int    // total number of times the test has run
+}
+
+// SelectTestsReq is the request for CategoriseTests
+type SelectTestsReq struct {
+	ForPastDays, // number of days data to consider for test selection
+	FirstRunOn, // number of days to consider for the first time the test is run
+	LastRunOn, // number of days to consider for the last time the test is run
+	SelectFromSuccessPct int //percentage of tests to be Selected for running from the successful test list sorted by number of runs
+	Cloud, // the cloud where the tests were run
+	Suite string // the test suite for which the selection is done
+}
+
+// NewDefaultSelectTestsReq returns a new SelectTestsReq with default values populated
+func NewDefaultSelectTestsReq(selectFromSuccessPct int, cloud, suite string) *SelectTestsReq {
+	return &SelectTestsReq{
+		ForPastDays:          defaultForPastDays,
+		FirstRunOn:           defaultFirstRunOn,
+		LastRunOn:            defaultLastRunOn,
+		SelectFromSuccessPct: selectFromSuccessPct,
+		Cloud:                cloud,
+		Suite:                suite,
+	}
+}
+
+// CategoriseTests returns the tests categorized based on the snowflake query
+// The tests are Selected by selector.go based on certain criteria:
+// 1. the number of time a test has been successfully running
+// 2. the test is new
+// 3. the test has not been run for a while
+// 4. a subset of the successful tests based on SelectTestReq.SelectFromSuccessPct
+// It returns all the tests. The selected tests have the value TestDetails.Selected as true
+func CategoriseTests(ctx context.Context, req *SelectTestsReq) ([]*TestDetails, error) {
+	db, err := getConnect(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = db.Close() }()
+	statement, err := db.Prepare(preparedQuery)
+	if err != nil {
+		return nil, err
+	}
+	// add the parameters in sequence
+	rows, err := statement.QueryContext(ctx,
+		req.ForPastDays*-1, fmt.Sprintf("%%%s - %s%%", suites[req.Suite], req.Cloud),
+		req.FirstRunOn*-1, req.LastRunOn*-1)
+	if err != nil {
+		return nil, err
+	}
+	// All the column headers
+	colHeaders, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+	// Will be used to read data while iterating rows.
+	colPointers := make([]interface{}, len(colHeaders))
+	colContainer := make([]string, len(colHeaders))
+	for i := range colPointers {
+		colPointers[i] = &colContainer[i]
+	}
+	// allTestDetails are all the tests that are obtained from snowflake query
+	allTestDetails := make([]*TestDetails, 0)
```

Comment:
Nit: IMO it's a bit confusing that `allTestDetails` initially means "selected tests" and only at the very end actually means "all tests". I suggest naming this `selectedTestDetails` (since it's the what this slice holds for most of the function), and then when returning, you can do `return append(selectedTests, skippedTests...)` which also conveys well the meaning that all tests are returned at the end.